Use VM name instead of Node name for disk operations with Flex VMSS

okushchenko opened the pull request:
#### What type of PR is this?

/kind bug

#### What this PR does / why we need it:

When using https://github.com/kubernetes-sigs/azuredisk-csi-driver with Flex VMSS, I observed that it fails to attach and detach disks from VMs. This is due to it trying to use k8s Node name instead of VM name when interacting with Azure APIs. This PR updates this logic to use a correct name.

#### Does this PR introduce a user-facing change?

```release-note
Fixed a bug that prevents disks from being attached and detached for VMs in a Flex VMSS
```

zmyzheng commented on pkg/provider/azure_controller_vmssflex.go:
```diff
@@ -38,7 +38,7 @@ func (fs *FlexScaleSet) AttachDisk(ctx context.Context, nodeName types.NodeName,
 	}
 
-	nodeResourceGroup, err := fs.GetNodeResourceGroup(vmName)
+	nodeResourceGroup, err := fs.GetNodeResourceGroup(*vm.Name)
```
CSI Driver is  using nodeName as the unique key to identify nodes/VMs. Changing to vm.name might cause issue here if external resource group is used. Can you change back to use vmName?

zmyzheng commented on pkg/provider/azure_controller_vmssflex.go:
```diff
@@ -97,20 +97,20 @@ func (fs *FlexScaleSet) AttachDisk(ctx context.Context, nodeName types.NodeName,
 	}()
 
-	klog.V(2).Infof("azureDisk - update(%s): vm(%s) - attach disk list(%s)", nodeResourceGroup, vmName, diskMap)
+	klog.V(2).Infof("azureDisk - update(%s): vm(%s) - attach disk list(%s)", nodeResourceGroup, *vm.Name, diskMap)
```
Can you change back to use vmName?

zmyzheng reviewed (commented):
Thank you so much for finding this issue. You are right that we should use VM name rather than nodeName (OS hostName) when updating VM, so line 102,109, 182, 189 should be fixed as you proposed in the PR. 

As for the rest lines, since CSI Driver is using nodename as the primary key to identify nodes/VMs, can we keep the way of using vmName? Only changing line 102,109, 182, 189 should be able to fix the issue

zmyzheng reviewed (commented):
Can you also update line 222 as well?

okushchenko commented:
@zmyzheng thanks for the quick review! I updated the PR with your suggestions and updated test cases for the `UpdateVM` method to account for the change that you proposed.

zmyzheng reviewed (approved):
/lgtm

okushchenko commented:
@zmyzheng I rebased and squashed the commits to satisfy the bots, feel free to modify this PR however you think is necessary to get it merged.

zmyzheng commented:
@feiskyer @andyzhangx could you help approve this PR?

nilo19 commented:
/lgtm
/approve

zmyzheng commented:
/retest

nilo19 commented:
/ok-to-test
/retest
